### PR TITLE
LPS-175089 Update testcase to check all the test cases that could run without the browser

### PR DESCRIPTION
**Pull request by Hijky**
**Background**
Update testcase to check all the test cases that could run without the browser

**Test Plan**
In all test cases with 
property testray.main.component.name = "Headless"; check the presence of @web-driver="true" and analyze if it is applied to all of the cases that can run without a browser being opened.
Remember to make sure TestCase.setUpPortalInstanceNoSelenium(); is present in setup, and PortalInstances.tearDownCPNoSelenium(); is present in tearDown.

**JIRA Link:**
https://issues.liferay.com/browse/LPS-175089

### CODE REVIEW

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:headless-functional:nocompile

**Comment by Hijky** [comment]
The failures are not related to my PR.

**Review by magjed4289 — COMMENTED** [review]
the purpose of this task was to check whether @web-driver="true" is applied in all testcase that do not use browser. I don't see any change related to this 🤔

**Comment by yanan** [comment]
> 

I've updated a few files, but I think it's better to update this once the refactor pr merged since there would be a few conflicts on noSelenium parameter. :smile:

**Comment by yanan** [comment]
> the purpose of this task was to check whether @web-driver="true" is applied in all testcase that do not use browser. I don't see any change related to this thinking
I've updated a few files, but I think it's better to update this once the refactor pr merged since there would be a few conflicts on noSelenium parameter. smile

**Comment by magjed4289** [comment]
yeah, also are we sure that the` @disable-webdriver = "true"` is everywhere where it could be? I remember seeing some cases that didn't have it and could, just a confirmation from @Hijky that it's been verified is enough :D

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:headless-functional:nocompile

**Comment by Hijky** [comment]
I have checked all of the component.name = "Headless" "Lima Headless" "Echo Headless", and updated them, and I found a problem in refactor [PR](https://github.com/liferay-headless/liferay-portal/pull/1041) and modified it.
**before**
Screenshot from 2023-03-08 14-19-46

**after**
Screenshot from 2023-03-08 14-21-01

File: modules/apps/object/object-rest-test/src/testFunctional/tests/SupportManyToManyRelationships.testcase [hunk 1]
```diff
@@ -13,19 +13,20 @@ definition {
...
 
 	tearDown {
-		if (${testPortalInstance} == "true") {
-			PortalInstances.tearDownCP();
-		}
-		else {
-			ObjectDefinitionAPI.modifyRelationship(
-				deletionType = "cascade",
-				objectName = "University");
+		ObjectDefinitionAPI.modifyRelationship(
```
Comment (yanan) [inline modules/apps/object/object-rest-test/src/testFunctional/tests/SupportManyToManyRelationships.testcase:16]: Thanks for adding tearDownCp/tearDownCPNoSelenium in some of the files. Just wondering if we use tearDownCP..., then all portal instance will be deleted, so I think we don't need to move the following from else condition. No sure if my understanding is correct, any ideas @magjed4289 @Hijky ?

File: modules/apps/headless/headless-admin-taxonomy/headless-admin-taxonomy-test/src/testFunctional/tests/ExposeErcForCategories.testcase [hunk 2]
```diff
@@ -19,13 +19,14 @@ definition {
...
 		if (${testPortalInstance} == "true") {
-			PortalInstances.tearDownCP();
+			PortalInstances.tearDownCPNoSelenium();
 		}
 		else {
 			JSONDepot.deleteDepot(depotName = "Test Depot Name");
 		}
 	}
 
+	@disable-webdriver = "true"
```
Comment (yanan) [inline modules/apps/headless/headless-admin-taxonomy/headless-admin-taxonomy-test/src/testFunctional/tests/ExposeErcForCategories.testcase:29]: :+1:

**Comment by yanan** [comment]
I've rebased to master and add more tests to disable the webdriver. The only concern on this pr is [the upper comment on tearDown part](https://github.com/liferay-headless/liferay-portal/pull/1058#discussion_r1131958782).

**Comment by yanan** [comment]
ci:test:headless-functional:nocompile

File: modules/apps/object/object-rest-test/src/testFunctional/tests/SupportManyToManyRelationships.testcase (hunk 1 above)
Comment (magjed4289) [reply modules/apps/object/object-rest-test/src/testFunctional/tests/SupportManyToManyRelationships.testcase:16]: @yanan if we use tearDownCP it should delete everything from the test but that is not always the case. also, the teardownCP would work in CI. Locally, if tests running without creating a new instance items need to be deleted separately.

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:headless-functional:nocompile

**Review by magjed4289 — APPROVED** [review]

**Comment by yanan** [comment]
ci:test:headless-functional:nocompile

**Comment by yanan** [comment]
ci:test:sf

**Comment by yanan** [comment]
Missing one previous commit to disable webdriver on more tests :smile: I've add more files and trigger our funcation tests cc @Hijky @magjed4289

**Comment by yanan** [comment]
No test failures related to the PR.

**Review by magjed4289 — APPROVED** [review]

**Comment by magjed4289** [comment]
LGTM! you can forward it! :) 🙏🏻

**Comment by yanan** [comment]
ci:forward

**Pull request closed without merging**